src/App.js

```diff
@@ -1,19 +1,56 @@
 import React from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+// import ChatEntry from './components/ChatEntry';
+import ChatLog from './components/ChatLog';
+import { useState } from 'react';
 
 const App = () => {
-  return (
+  const initialCopy = chatMessages.map((chat) => {
+    return {...chat};
+  });
+  const[chatData, setChatData] = useState(initialCopy);
+  const [likedCount, setLikedCount] = useState(0);
+
+    const heartWidget = likedCount !== 0 ? `${likedCount} ❤️s`: '';
+    // create a new list of entries with updated likes 
+    const updateLikes = (chatEntryId, updatedLikes) => {
+      let newLikeCount = likedCount;
+      const newChatData = chatData.map((chatEntry) => {
+        if (chatEntry.id !== chatEntryId) {
+          return chatEntry;
+        } else {
+          if (updatedLikes) {
+            newLikeCount++;
+          } else {
+            newLikeCount--;
+          }
+          const newChat = {
+            ...chatEntry,
+            liked: updatedLikes,
+          };
+          return newChat;
+        }
+      });
+    setChatData(newChatData);
+    setLikedCount(newLikeCount);
+    };
```

Comment:
Here's a few ways we can refactor your code:

1) We can pass in an anonymous function to the state function `setLikedCount` so that we won't need to update the variable `newLikeCount`.  ```setLikedCount(prev => prev + 1);``` where `prev` represents the previous value of likedCount.


```javascript
    const updateLikes = (chatEntryId, updatedLikes) => {
      const newChatData = chatData.map((chatEntry) => {
        if (chatEntry.id !== chatEntryId) {
          return chatEntry;
        } else {
          if (updatedLikes) {
            setLikedCount(prev => prev + 1);
          } else {
             setLikedCount(prev => prev - 1);
          }
          
          const newChat = {
            ...chatEntry,
            liked: updatedLikes,
          };
          return newChat;
        }
      });
    setChatData(newChatData);
    };
```